.aws/deploy/backend-task-definition.prod.json

```diff
@@ -0,0 +1,221 @@
+        { "name": "NODE_ENV", "valueFrom": "PROD_NODE_ENV" },
+        { "name": "OTP_EXPIRY", "valueFrom": "PROD_OTP_EXPIRY" },
+        { "name": "OTP_SECRET", "valueFrom": "PROD_OTP_SECRET" },
+        {
+          "name": "OUTGOING_QUEUE_URL",
+          "valueFrom": "PROD_OUTGOING_QUEUE_URL"
+        },
+        { "name": "POSTMAN_API_KEY", "valueFrom": "PROD_POSTMAN_API_KEY" },
+        {
+          "name": "POSTMAN_SMS_CRED_NAME",
+          "valueFrom": "PROD_POSTMAN_SMS_CRED_NAME"
+        }
+      ],
+      "logConfiguration": {
+        "logDriver": "awslogs",
+        "options": {
+          "awslogs-group": "/aws/elasticbeanstalk/cms-backend-prod-node18/var/log/web.stdout.log",
+          "awslogs-region": "ap-southeast-1",
+          "awslogs-stream-prefix": "ecs"
+        }
+      }
+    },
+    {
+      "name": "dd-agent",
+      "image": "public.ecr.aws/datadog/agent:latest",
+      "portMappings": [
+        {
+          "containerPort": 8126,
+          "hostPort": 8126,
+          "protocol": "tcp"
+        }
+      ],
+      "essential": true,
+      "environment": [
+        {
+          "name": "TZ",
+          "value": "Asia/Singapore"
+        },
+        {
+          "name": "DD_APM_NON_LOCAL_TRAFFIC",
+          "value": "true"
+        },
+        {
+          "name": "ECS_FARGATE",
+          "value": "true"
+        },
+        {
+          "name": "DD_APM_ENABLED",
+          "value": "true"
+        },
+        {
+          "name": "DD_SITE",
+          "value": "datadoghq.com"
+        },
+        {
+          "name": "DD_ENV",
+          "value": "prod"
+        },
+        {
+          "name": "DD_SERVICE",
+          "value": "isomer"
+        },
+        {
+          "name": "DD_TAGS",
+          "value": "service:isomer"
+        },
+        {
+          "name": "DD_AGENT_MAJOR_VERSION",
+          "value": 7
+        },
+        {
+          "name": "DD_LOGS_INJECTION",
+          "value": true
+        },
+        {
+          "name": "DD_TRACE_STARTUP_LOGS",
+          "value": true
+        },
+        {
+          "name": "DD_API_KEY",
+          "valueFrom": "<DD_API_KEY>"
+        }
+      ],
+      "mountPoints": [],
+      "volumesFrom": [],
+      "secrets": [],
+      "logConfiguration": {
+        "logDriver": "awslogs",
+        "options": {
+          "awslogs-group": "isomer-infra/ecs/dd-agent",
+          "awslogs-region": "ap-southeast-1",
+          "awslogs-stream-prefix": "ecs"
+        }
+      }
+    }
+  ],
+  "family": "isomer-infra",
+  "networkMode": "awsvpc",
+  "volumes": [
+    {
+      "name": "efs",
+      "efsVolumeConfiguration": {
+        "fileSystemId": "<EFS_FILE_SYSTEM_ID>",
+        "rootDirectory": "/efs"
+      }
+    }
+  ],
+  "placementConstraints": [],
+  "runtimePlatform": {
+    "operatingSystemFamily": "LINUX"
+  },
+  "requiresCompatibilities": ["FARGATE"],
+  "taskRoleArn": "arn:aws:iam::<AWS_ACCOUNT_ID>:role/isomer-infra-github-oidc-role-16ea937",
```

Comment:
is this role managed by us or pulumi? will this hash change?

Reply:
we only ever create the role once and don't recreate it so idt it iwll change.

Reply:
how come the role has no <env> prefix ah? do we use the same roles for the different env? understanding here is that we should be having at least 2 roles, `stg` + `prod`?

Reply:
no, the idp is **one per aws account**. because our staging and prod are all on 1 account, **even if we prefix**, it wouldn't be able to create another idp.